### DISCUSSION TITLE
Empty username and password with plugin rabbitmq_auth_backend_http

### DISCUSSION THREAD

**Discussion by janiu-001** [rabbitmq/rabbitmq-server#5731] [Other]
Hi expert, 

#### 1. I enabled the plugin rabbitmq_auth_backend_http to verify the client, following is my configuration:
  [
    {rabbit, [{auth_backends, [rabbit_auth_backend_http]}]},
    {rabbitmq_auth_backend_http,
      [{http_method,   post},
       {user_path,     "http://host.docker.internal:3001/users"},
       {vhost_path,    "http://host.docker.internal:3001/users"},
       {resource_path, "http://host.docker.internal:3001/users"},
       {topic_path,    "http://host.docker.internal:3001/users"}]}
  ].

#### 2. And i set the user name and password with connect option:

<img width="864" alt="image" src="https://user-images.githubusercontent.com/19970581/189246510-7f7face0-c823-4eaa-8d8a-588765f3b69b.png">

#### 3. But I could not get the username and password from body, the body is empty:

<img width="701" alt="image" src="https://user-images.githubusercontent.com/19970581/189246645-41b9e804-d003-4b90-b87e-efcb210858d5.png">

**Comment by michaelklishin** [comment]
```
{user_path, "http://host.docker.internal:3001/users"},
{vhost_path, "http://host.docker.internal:3001/users"},
{resource_path, "http://host.docker.internal:3001/users"},
```

looks like a typo. You have an identical URL for all three endpoints.

Start with verifying [effective configuration](https://www.rabbitmq.com/configure.html#verify-configuration-effective-configuration) and consider using the "new" (has been around for a few years now) [configuration format](https://www.rabbitmq.com/configure.html#config-file) which the HTTP authN backend supports:

``` ini
auth_backends.1 = http
auth_http.http_method   = post
auth_http.user_path     = http://some-server/auth/user
auth_http.vhost_path    = http://some-server/auth/vhost
auth_http.resource_path = http://some-server/auth/resource
auth_http.topic_path    = http://some-server/auth/topic
```
